app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/home/details/BirthdayDetailsScreen.kt

```diff
@@ -165,6 +173,23 @@ fun BirthdayDetailsScreen(
                     )
                 }
             }
+            if (showGenerateMessageDialog) {
+                GenerateMessageDialog(
+                    message = birthdayMessage,
+                    onValueChange = { message -> birthdayDetailsViewModel.setBirthdayMessage(message) },
+                    onDismissRequest = { showGenerateMessageDialog = false },
+                    onConfirmButtonClick = {
+                        val sendIntent = Intent().apply {
+                            action = Intent.ACTION_SEND
+                            putExtra(Intent.EXTRA_TEXT, birthdayMessage)
+                            type = "text/plain"
+                        }
+                        val shareIntent = Intent.createChooser(sendIntent, null)
+                        ContextCompat.startActivity(context, shareIntent, null)
```

Comment:
Create an extension for this Intent. There can be cases when you would need this in other places too.